streaming/streaming.go

```diff
@@ -304,22 +304,54 @@ func (c *Conn) receivedOrderBook(ob orderBook) error {
 }
 
 func (c *Conn) receivedUpdate(u Update) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	valid, err := c.validateUpdate(u)
+	if err != nil {
+		return err
+	}
+
+	// If update is not valid, ignore
+	if !valid {
+		return nil
```

Comment:
What sort of update would we get that we expect not to be valid? Should we rather disconnect if we get an invalid update?

Reply:
The current behaviour is as follows:

If current sequence is 0 -> skip event (orderbook is not initialised yet)
if received update sequence is less the current sequence -> skip event (it is an old event and no longer relevant)
if current sequence +1 != received update sequence -> Out of sequence event received, disconnect

validateUpdate returns a bool(valid) and error. If !valid, we can skip. If error is not nil, disconnect. That way we maintain the current behaviour.

Reply:
We can change it to rather disconnect if we receive an old event but I can't think of a scenario where this would happen unless there is a bug in luno's marketstream. IMO we should stay with how it currently is.